src/DIRAC/Interfaces/API/Dirac.py

```diff
@@ -1675,6 +1675,10 @@ def deleteJob(self, jobID):
             return ret
         jobID = ret["Value"]
 
+        res = JobStatus.checkJobStateTransition(jobID, JobStatus.DELETED)
```

Comment:
> If running, these jobs will be first killed.

I don't think it will work as expected:
```
>>> JobsStateMachine('Running').getNextState('Deleted')
{'OK': True, 'Value': 'Running'}
```

Reply:
I added the `checkJobStateTransition` that would return `{'OK': False}` in this case.